src/main/java/org/wso2/securevault/definition/TrustKeyStoreInformation.java

```diff
@@ -38,10 +41,13 @@ public TrustManagerFactory getTrustManagerFactoryInstance() {
                 log.debug("Creating a TrustManagerFactory instance");
             }
             KeyStore trustStore = this.getTrustStore();
-            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
-                    TrustManagerFactory.getDefaultAlgorithm());
+            TrustManagerFactory trustManagerFactory;
+            if (MiscellaneousUtil.getPreferredJceProvider() != null) {
```

Comment:
Do we need to use PKIX for FIPS provider?

Reply:
Yes, using this document as a reference: [sample](https://downloads.bouncycastle.org/fips-java/docs/BC-FJA-%28D%29TLSUserGuide-2.1.20.pdf)